Unity-Technologies/Graphics
# Add Performance framework package and HDRP performance tests

### Purpose of this PR
Original PR: https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/6356
- Add the graphic performance framework package, based on the performance testing package: https://docs.unity3d.com/Packages/com.unity.test-framework.performance@1.0/manual/index.html
- Add the implementation of HDRP tests for graphic performance.

### Testing status
This framework helps you to setup a performance testing which includes: 
- Frame timings (based on profilingSampler)
- Memory profiling (based on asset profiling)
- Build time (shader compilation, total build time, variant count after stripping per platform)
- Build size (compiled and compressed shader size, Compute shader size ...)
- VGPR / SGPR for shader, ComputeShader, ShaderGraphs, and materials

Check out the latest documentation: https://github.com/Unity-Technologies/ScriptableRenderPipeline/blob/HDRP/performance-testing/com.unity.testing.graphics-performance/Documentation~/com.unity.testing.graphics-performance.md

Here's the result of a performance test in the editor:
![](https://user-images.githubusercontent.com/6877923/77452606-6b6e6200-6df6-11ea-8e1e-aa50206613b0.png)

Also made yamato pipeline so performance tests are automatically run every day
![](https://user-images.githubusercontent.com/6877923/77452581-63aebd80-6df6-11ea-897a-e5c42c45eb46.png)

Here's the performance dashboard that we made for HDRP, other teams that wants to use it will need to build their own one to visualize the performance data.
![](https://user-images.githubusercontent.com/6877923/77452637-76c18d80-6df6-11ea-8996-09038685afa7.png)
https://grafana.internal.unity3d.com/d/cLLln6UWk/hdrp-performances?orgId=1&from=now-7d&to=now

In-scene test settings description:
![image](https://user-images.githubusercontent.com/6877923/78264182-22f42a00-7503-11ea-8ca9-4961b249a95c.png)
